src/utilities/git.jl

```diff
@@ -1,5 +1,86 @@
-git_checkout(branch::AbstractString) = run(`git checkout $(branch)`)
+const COMPATHELPER_GIT_COMMITTER_NAME = "CompatHelper Julia"
+const COMPATHELPER_GIT_COMMITTER_EMAIL = "compathelper_noreply@julialang.org"
 
-function git_clone(url::AbstractString, local_path::AbstractString)
-    return run(`git clone $(url) $(local_path)`)
+function get_git_name_and_email(; env=ENV)
+    name = if haskey(env, "GIT_COMMITTER_NAME")
+        env["GIT_COMMITTER_NAME"]
+    else
+        COMPATHELPER_GIT_COMMITTER_NAME
+    end
+
+    email = if haskey(env, "GIT_COMMITTER_EMAIL")
+        env["GIT_COMMITTER_EMAIL"]
+    else
+        COMPATHELPER_GIT_COMMITTER_EMAIL
+    end
+
+    return name, email
+end
+
+git_checkout(branch::AbstractString) = run(`git checkout $branch`)
+
+function git_add(; items::AbstractString=".", flags::AbstractString="")
```

Comment:
I guess not. I think I just unintentionally try to make things more generic than they need to be. This could be changed to just always add everything